csrc/device_lower/utils.cpp

```diff
@@ -781,21 +781,27 @@ bool hasBlockSync(const Expr* expr, const ThreadPredicateMap& pred_map) {
     return false;
   }
 
-  if (!(ir_utils::isReductionOp(expr) || expr->isA<BroadcastOp>() ||
-        expr->isA<kir::GridBroadcast>())) {
-    return false;
-  }
-
   // GroupedReductionOp can have multiple output TVs, but they must be
   // parallelized in the same way, so just checking one of them is enough.
   auto tv = ir_utils::getTvOutput(expr);
 
-  if (tv->hasBlockReduction() || tv->hasGridReduction()) {
+  if (ir_utils::isReductionOp(expr) &&
```

Comment:
The change here only adds the condition for `ArgsortOp` and `TopKOp`, except this function previously returns false for `GridBroadcast`, which was a bug. We almost always do a grid broadcast as part of a preceding grid reduction, so I believe there's no test case that generates a grid broadcast call. In fact, I couldn't reproduce anything.